contrib/raftexample/raft.go

```diff
@@ -247,7 +302,76 @@ func (rc *raftNode) stopHTTP() {
 	<-rc.httpdonec
 }
 
+func (rc *raftNode) applySnapshot(snapshotToSave raftpb.Snapshot) {
+	if raft.IsEmptySnap(snapshotToSave) {
+		return
+	}
+
+	log.Printf("applying snapshot at index %d", rc.snapshotIndex)
+	defer log.Printf("finished applying snapshot at index %d", rc.snapshotIndex)
+
+	if snapshotToSave.Metadata.Index <= rc.appliedIndex {
+		log.Fatalf("snapshot index [%d] should > progress.appliedIndex [%d] + 1", snapshotToSave.Metadata.Index, rc.appliedIndex)
+	}
+
+	spath := filepath.Join(rc.snapdir, fmt.Sprintf("%016x-%016x.snap", snapshotToSave.Metadata.Term, snapshotToSave.Metadata.Index))
```

Comment:
@xiang90 

We rename it here for `(s *kvstore) readCommits` that doesn't know the snapshot metadata; it only receives snapshot bytes from `commitC` channel. So there should be a snapshot file without index in snapshot directory.

Maybe I am wrong. Any suggestion?

Thanks!